### PR TITLE
New: CNS configuration API

### PR DESCRIPTION
**Pull request by ericrpowers**
## Description
Per https://redlock.atlassian.net/browse/CNS-1832, we want an API endpoint like the following: `/cnsconfig` which will eventually store all PC configuration for a namespace. To start:
- The object has the attributes EnableNetEffectivePermissions and EnableNetworkSecurity.
- The object has the attribute `Namespace` which will be used to have only one config per namespace.

### CODE REVIEW

File: specs/cnsconfig.spec [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  aliases:
+  - pcc
+  get:
+    description: Retrieve the CNS configuration with the given ID.
+  update:
+    description: Updates the CNS configuration with the given ID.
+  delete:
+    description: Deletes the CNS configuration with the given ID.
+  extends:
+  - '@identifiable-stored'
```
Comment (primalmotion) [inline specs/cnsconfig.spec:18]: @namespaced is at least required, maybe we want @timeable to etc.. please revise this

**Review by primalmotion — CHANGES_REQUESTED** [review]

File: specs/cnsconfig.spec [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    stored: true
+
+  - name: enableNetworkSecurity
+    description: If `true` network security feature is enabled.
+    type: boolean
+    exposed: true
+    stored: true
+
+  - name: key
+    description: The unique key of the configuration.
```
Comment (t00f) [inline specs/cnsconfig.spec:40]: What is the unique key here if we have an identifiable? What will it store?

**Review by t00f — APPROVED** [review]
Looks good! Just one minor comment

File: specs/cnsconfig.spec (hunk 2 above)
Comment (ericrpowers) [reply specs/cnsconfig.spec:40]: Per the description above:
```
The object has an attribute "key" that is the namespace for uniqueness.
```
The idea is to have one and only one config for a namespace. Thinking on this more, with the recent extension `@namespaced` added, we can just leverage the `Namespace` attribute to achieve this.

**Comment by dstiliadis** [comment]
I don't think the namespaced extension is enough. The problem is that it is not guarantee uniqueuness. We do need a key that is unique in the DB that is equal to the namespace.

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "87b931c1ac3da0b6a2f2606d9fdd18c9b7f30591",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "4f224352b44ad381cc9874178c7dda5de8f74b8f",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "f2322f3582fc1c97d95aafa0dd161e34d3ae30d7",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "17129105df1d99003c1c5a8f17ad258bf7d46e81",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "fc58d29ea622cca38959347766a25f884403d5aa",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "5e8a32bc7f675dede21f786aa5e1699a2b599d1c",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "fc58d29ea622cca38959347766a25f884403d5aa",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "fc58d29ea622cca38959347766a25f884403d5aa",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "49035688aded9621a2d76e16e5db5822639e44e6",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "766993578b5da30791de40b4f3853f4c537647f3",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "fc58d29ea622cca38959347766a25f884403d5aa",
    "pipeline": "master"
  }
]
```

**Review by primalmotion — DISMISSED** [review]

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "fa6fdfa3f2765d2ee25fc1b640eb3d2cf367a93c",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "6142547121228a0aa370f4044eae134472c3842e",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "fa6fdfa3f2765d2ee25fc1b640eb3d2cf367a93c",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "ebac1a53e8af6fe14d42d4f37f8edaae31fd62e4",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "6142547121228a0aa370f4044eae134472c3842e",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "prisma-cloud-configuration",
    "component": "backend",
    "pr-id": "1180",
    "commit-sha": "c43db88ca6ebcce363cd77754f208582e3141869",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "gaia",
    "pr-id": "649",
    "commit-sha": "2fdb6d8308127a02d4210aef3bad5cbbecc494f8",
    "pipeline": "master"
  },
  {
    "project": "prisma-cloud-configuration",
    "component": "prismacloud",
    "pr-id": "33",
    "commit-sha": "49b53b2b580adbd0c78160d4dd63ee220a02af92",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]
LGTM

**Review by primalmotion — APPROVED** [review]

**Pull request merged**